samples/snippets/sample_firewall.py

```diff
@@ -0,0 +1,159 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from typing import Iterable
+
+# [START compute_firewall_list]
+# [START compute_firewall_create]
+# [START compute_firewall_patch]
+# [START compute_firewall_delete]
+import google.cloud.compute_v1 as compute_v1
+# [END compute_firewall_delete]
+# [END compute_firewall_patch]
+# [END compute_firewall_create]
+# [END compute_firewall_list]
+
+
+# [START compute_firewall_list]
+def list_firewall_rules(project_id: str) -> Iterable:
+    """
+    Return a list of all the firewall rules in specified project.
+
+    Args:
+        project_id: project ID or project number of the Cloud project you want to use.
+
+    Returns:
+    A flat list of all the firewall rules defined for given project.
+    """
+    firewall_client = compute_v1.FirewallsClient()
+    firewalls_list = firewall_client.list(project=project_id)
+    return firewalls_list
+# [END compute_firewall_list]
+
+
+def print_firewall_rule(project_id: str, firewall_rule_name: str):
+    firewall_client = compute_v1.FirewallsClient()
+    print(firewall_client.get(project=project_id, firewall=firewall_rule_name))
+
+
+# [START compute_firewall_create]
+def create_firewall_rule(project_id: str, firewall_rule_name: str):
+    """
+    Creates a simple firewall rule allowing for incoming HTTP and HTTPS access from the entire Internet.
+
+    Args:
+        project_id: project ID or project number of the Cloud project you want to use.
+        firewall_rule_name: name of the rule that is created.
+    """
+    firewall_rule = compute_v1.Firewall()
+    firewall_rule.name = firewall_rule_name
+    firewall_rule.direction = compute_v1.Firewall.Direction.INGRESS
+
+    tcp_80_443_allowed = compute_v1.Allowed()
+    tcp_80_443_allowed.I_p_protocol = "tcp"
+    tcp_80_443_allowed.ports = ["80", "443"]
+
+    firewall_rule.allowed = [tcp_80_443_allowed]
+    firewall_rule.source_ranges = ["0.0.0.0/0"]
+    firewall_rule.network = "global/networks/default"
```

Comment:
We should probably mention in the comment's that this rule is created in the default network, another option is to make it available as an argument with this as default value